extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/DefaultChunkWriterFactory.java

```diff
@@ -0,0 +1,1363 @@
+                            } else {
+                                // days then millis
+                                out.writeInt((int) value.toDays());
+                                out.writeInt((int) (value.getNano() / nsPerMs));
+                            }
+                        });
+
+            default:
+                throw new IllegalArgumentException("Unexpected interval unit: " + intervalType.getUnit());
+        }
+    }
+
+    private static ChunkWriter<ObjectChunk<Period, Values>> intervalFromPeriod(
+            final BarrageTypeInfo<Field> typeInfo) {
+        /*
+         * A "calendar" interval which models types that don't necessarily have a precise duration without the context
+         * of a base timestamp (e.g. days can differ in length during day light savings time transitions). All integers
+         * in the types below are stored in the endianness indicated by the schema.
+         *
+         * @formatter:off
+         * YEAR_MONTH:
+         * Indicates the number of elapsed whole months, stored as 4-byte signed integers.
+         *
+         * DAY_TIME:
+         * Indicates the number of elapsed days and milliseconds (no leap seconds), stored as 2 contiguous 32-bit signed
+         * integers (8-bytes in total).
+         *
+         * MONTH_DAY_NANO:
+         * A triple of the number of elapsed months, days, and nanoseconds. The values are stored
+         * contiguously in 16-byte blocks. Months and days are encoded as 32-bit signed integers and nanoseconds is
+         * encoded as a 64-bit signed integer. Nanoseconds does not allow for leap seconds.
+         * @formatter:on
+         *
+         * Note: Period does not handle the time portion of DAY_TIME and MONTH_DAY_NANO. Arrow stores these in
+         * PeriodDuration pairs.
+         */
+        final ArrowType.Interval intervalType = (ArrowType.Interval) typeInfo.arrowField().getType();
+        switch (intervalType.getUnit()) {
+            case YEAR_MONTH:
+                return new IntChunkWriter<>(ObjectChunk::isNull, ObjectChunk::getEmptyChunk, (chunk, ii) -> {
+                    final Period value = chunk.get(ii);
+                    return value == null ? QueryConstants.NULL_INT : value.getMonths() + value.getYears() * 12;
+                }, typeInfo.arrowField().isNullable());
+            case DAY_TIME:
+                return new FixedWidthChunkWriter<>(ObjectChunk::isNull, ObjectChunk::getEmptyChunk, Integer.BYTES * 2,
+                        false, typeInfo.arrowField().isNullable(),
+                        (out, chunk, offset) -> {
+                            final Period value = chunk.get(offset);
+                            if (value == null) {
+                                out.writeInt(0);
+                                out.writeInt(0);
+                            } else {
+                                // days then millis
+                                out.writeInt(value.getDays());
```

Comment:
So, I'm not 100% certain we can be much smarter. In that, the number of days in a month (and year) are not exactly fixed.

Apache Arrow's implementation does not convert between time units and expects them to be separate. They don't offer `set` methods that take PeriodDuration / Duration but rather those methods take one input parameter for each time unit. They do offer `getObject` calls, but they also assume that the time unit values are separate.

AFAICT, this is the behavior that Arrow intends. Since the Vectors do not hold PeriodDuration/Duration, it seems impossible to provide an object to Arrow that won't fit in the designated wire-format/vector-type.